src/fileoverview_comment_transformer.ts

```diff
@@ -0,0 +1,77 @@
+import * as ts from 'typescript';
+import * as jsdoc from './jsdoc';
+
+/**
+ * A set of JSDoc tags that mark a comment as a fileoverview comment. These are recognized by other
+ * pieces of infrastructure (Closure Compiler, module system, ...).
+ */
+const FILEOVERVIEW_COMMENT_MARKERS: ReadonlySet<string> =
+    new Set(['fileoverview', 'externs', 'modName', 'mods', 'pintomodule']);
+
+/**
+ * A transformer that ensures the emitted JS file has an \@fileoverview comment that contains a
+ * \@suppress {checkTypes} annotation by either adding or updating an existing comment.
+ */
+export function transformFileoverviewComment(context: ts.TransformationContext) {
+  return (sf: ts.SourceFile) => {
+    let comments: ts.SynthesizedComment[] = [];
+    // Use trailing comments because that's what transformer_util.ts creates (i.e. by convention).
+    if (sf.statements.length) comments = ts.getSyntheticTrailingComments(sf.statements[0]) || [];
```

Comment:
you could also check that the statement is a `NonEmittedStatement`, i.e. enforcing that the original user did add an extra newline between the comment and the first statement.
E.g. match this:
```
/** @fileoverview **/

class X {}
```

But not this:
```
/** @fileoverview **/
class X {}
```

Reply:
Done.